distributed/semaphore.py

```diff
@@ -67,16 +71,23 @@ def __init__(self, scheduler):
 
         self.scheduler.extensions["semaphores"] = self
 
+        # {metric_name: {semaphore_name: metric}}
+        self.metrics = {
+            "acquire_total": defaultdict(int),  # counter
+            "release_total": defaultdict(int),  # counter
+            "time_to_acquire_lease": defaultdict(list),  # summary-like
```

Comment:
this list is unbound and might pose a problem. I suggest to either use a limited `deque` or even better, use a `tdigest` since essentially we're interested in a histogram, aren't we?

Reply:
Yes, we are. Thanks for the suggestion. Will change this accordingly